app/controllers/application_controller/compare.rb

```diff
@@ -1851,4 +1849,53 @@ def section_field_compare_values(view, section, field, base_val)
       unset_same_flag unless fld.nil? || base_val == fld[:_value_]
     end
   end
+
+  def validate_name(name)
+    name&.starts_with?('xx-group')
+  end
+
+  # Method to get the selection names from :all_checked and :id params.
+  # individual selected item suggests that the variable name contains ':'
+  # else, the child_names are retrived from the selected_parent item.
+  def selection_names(name)
+    if validate_name(name)
+      count = name.scan(/xx-group/).count
+      count == 1 ? child_names(name) : [name.split(':')[1]]
+    end
+  end
+
+  # Method to get the child names of a selected parent item.
+  # session[:miq_sections] contains all tree names and the keys are filters using the selected group_name[1].
+  def child_names(name)
+    child_names = []
+    group_name = name.split('_')
+    if group_name[1]
+      child_sections = session[:miq_sections]&.select { |_key, value| value[:group] == group_name[1] }
+      if child_sections && child_sections.length > 1
+        child_names = child_sections.keys.map(&:to_s)
+      end
+    end
+    child_names
+  end
+
+  # Method to add or remove selections if an id is present.
+  def update_selections(param_id, selections, check)
+    if param_id
+      names = selection_names(CGI.unescape(param_id))
+      if check.match?(/(true|false)/)
+        check == 'true' ? add_selection(names, selections) : remove_selections(names, selections)
+      end
+    end
+    selections
+  end
+
+  # Method to add names to the selections.
+  def add_selection(names, selections)
```

Comment:
Super-super minor, but I think think should be named `add_selections` to be consistent with `remove_selections`.

```suggestion 
  def add_selections(names, selections)
```

Reply:
Also super-super minor, but since this modifies the incoming parameter, we should put a `!` as Ruby convention to show it's a "dangerous" method.

```suggestion
  def add_selections!(names, selections)
```

Same below with `remove_selections` -> `remove_selections!`